src/app/main/component/ubs/components/ubs-order-details/ubs-order-details.component.ts

```diff
@@ -266,24 +267,31 @@ export class UBSOrderDetailsComponent extends FormBaseComponent implements OnIni
 
   addCertificate(): void {
     this.additionalCertificates.push(this.fb.control('', [Validators.minLength(8), Validators.pattern(/(?!0000)\d{4}-(?!0000)\d{4}/)]));
-    this.addCert = false;
+    if (arguments.length === 0) {
+      this.addCert = false;
+    }
+  }
+
+  private clearAdditionalCertificate(i) {
+    this.additionalCertificates.removeAt(i);
+    this.certStatuses.splice(i, 1);
+    this.calculateCertificates(this.certificates);
   }
 
-  deleteCertificate(i): void {
+  deleteCertificate(i: number): void {
```

Comment:
it would be better to use index instead of i

Reply:
![image](https://user-images.githubusercontent.com/43093994/124005706-0edaf900-d9e2-11eb-8ccc-1ea354f2a6fe.png)